### PR TITLE
[SKILLONOMY-189] Course news tab

### PR DESCRIPTION
**Pull request by vlasovmichael**
**Youtrack:** https://youtrack.raccoongang.com/issue/SKILLONOMY-189

**Changes:**
* New tab "News"
* New field (title) for course updates in the cms
* Course program added to the News page

**Notes:**
* NEWS_ENABLED setting must be True
* New tab is default for the new courses and appears for existing
courses after changing any advanced settings in the cms

**Updated HTML layout**

pr in the theme https://github.com/raccoongang/edx-theme/pull/1905

### CODE REVIEW

File: cms/static/js/views/course_info_update.js [hunk 1]
```diff
@@ -218,6 +226,8 @@ define(['codemirror',
                 else {
                     $(this.dateEntry(event)).val('MM/DD/YY');
                 }
+                this.$codeMirrorTitle = CourseInfoHelper.editWithCodeMirror(
+                targetModel, 'content_title', self.options.base_asset_url, $textAreaTitle.get(0));
```
Comment (OPersian) [inline cms/static/js/views/course_info_update.js]: is something wrong with indentation here?

File: common/lib/xmodule/xmodule/tabs.py [hunk 2]
```diff
@@ -390,7 +390,7 @@ def initialize_default(course):
         course.tabs.extend([
             CourseTab.load('textbooks'),
             discussion_tab,
-            CourseTab.load('wiki'),
+            CourseTab.load('news'),
```
Comment (OPersian) [inline common/lib/xmodule/xmodule/tabs.py]: I still think that it's better to not replace the old tabs with new ones, but create new tabs separately, irrespectively of old tabs. But it's OK)

File: lms/djangoapps/course_news/templates/course_news/course-updates-fragment.html [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                                <div class="course-news__title">
+                                    ${HTML(update.get("content_title"))}
+                                </div>
+                                % if not update.get("is_error"):
+                                    <div class="course-news__date">${update.get("date")}</div>
+                                % endif
+                                <div class="course-news__text">${HTML(update.get("content"))}</div>
+                                <div class="course-news__actions">
+                                    <button>
+                                        Подробнее
```
Comment (OPersian) [inline lms/djangoapps/course_news/templates/course_news/course-updates-fragment.html]: The original term will always be in Russian, not being marked for i18n

If intended, then it's OK

File: lms/djangoapps/course_news/views.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        return ordered_updates
+
+    @classmethod
+    def has_updates(self, request, course):
+        return len(self.get_ordered_updates(request, course)) > 0
+
+    @classmethod
+    def get_plain_html_updates(self, request, course):
+        """
+        Returns any course updates in an html chunk. Used
```
Comment (OPersian) [inline lms/djangoapps/course_news/views.py]: This and other `CourseNewsFragmentView`'s methods duplicate `CourseUpdatesFragmentView` methods' logic. Maybe it's worth inheriting `CourseUpdatesFragmentView`, not `EdxFragmentView`.   On the other hand, it'd be confusing. I'm not sure :]

Comment (dyudyunov) [reply lms/djangoapps/course_news/views.py:93]: TBD

File: lms/djangoapps/course_news/templates/course_news/course-updates-fragment.html (hunk 3 above)
Comment (dyudyunov) [reply lms/djangoapps/course_news/templates/course_news/course-updates-fragment.html:71]: Last time I spoke to the client he asked if we could change the site to be displayed only in Russian, so it's not a problem I think

File: common/lib/xmodule/xmodule/tabs.py (hunk 2 above)
Comment (dyudyunov) [reply common/lib/xmodule/xmodule/tabs.py:393]: TBD

File: cms/static/js/views/course_info_update.js (hunk 1 above)
Comment (dyudyunov) [reply cms/static/js/views/course_info_update.js]: Fixed, thanks!

**Pull request merged**